draft-kazuho-quic-authenticated-handshake.md

```diff
@@ -348,24 +348,33 @@ the hidden server, in addition to those required by the Encrypted SNI
 extension:
 
 * hmac_key
-* spare Connection IDs
 * ODCID
 
 Both the fronting server and the hidden server need access to the hmac_key to
 authenticate the Initial packets.  However, because the key is derived from
 the shared DH secret of ESNI, it is not necessarily available to the hidden
 server.
 
-Access to spare Connection IDs is mandatory to support clients migrating to
-different addresses.  The fronting server, without access to the 1-RTT QUIC
-frames being exchanged, cannot track the mapping of the Connection IDs that
-change mid-connection.
-
 ODCID is necessary to decrypt an Initial packet sent in response to a Retry.
 However, the value is typically available only to the server that generates
 the Retry.  The fronting server and the hidden server need to exchange the
 ODCID, or provide the secret for extracting the ODCID from a Retry token.
 
+Additionally, the following properties need to be exchanged when providing
+support for connection migration:
+
+* client's address tuple
+* spare Connection IDs
+
+The fronting server needs to forward the source IP address and port of the
+incoming packets to the hidden server, so that the hidden server can recognize
+the migration of the client and respond as required by QUIC version 1.
+
+The fronting server and the hidden server also need to share the spare
+Connection IDs being issued to the client using NEW_CONNECTION_ID frames, so
+that the fronting server can continue forwarding packets to the correct hidden
+server when the client starts using a new connection ID mid-connection.
+
 # Security Considerations
```

Comment:
Looks good. But we are close to specifying how to write a proxy...

Reply:
Good point. I've removed statements that discuss requirements not specific to QUIC-AH, and clarified the scope of the section. PTAL at 87a20ba.